packit/api.py

```diff
@@ -362,6 +363,129 @@ def update_source_git(self, revision_range: str):
                     f"Commit {commit} had no changes to be applied, skipping it."
                 )
 
+    @staticmethod
+    def _show_status(
+        sg_range_start: Optional[str], dg_range_start: Optional[str]
+    ) -> None:
+        """Outputs the synchronization status.
+
+        Args:
+            sg_range_start: Hash of the first commit in source-git that has
+                not been synchronized to dist-git yet. None if dist-git is
+                up-to-date.
+            dg_range_start: Hash of the first commit in dist-git that has
+                not been synchronized to source-git yet. None if source-git
+                is up-to-date.
+        """
+        if sg_range_start:
+            logger.info(
+                f"Dist-git status: source-git commit range {sg_range_start}.. must be synchronized."
+            )
+        else:
+            logger.info("Dist-git status: up-to-date")
+
+        if dg_range_start:
+            logger.info(
+                f"Source-git status: dist-git commit range {dg_range_start}.. must be synchronized."
+            )
+        else:
+            logger.info("Source-git status: up-to-date")
+
+        if not sg_range_start and not dg_range_start:
+            logger.info("Source-git and dist-git are fully synchronized.")
+
+    def _get_latest_commit_update_pair(self) -> Tuple[str, str]:
+        """Finds the latest pair of commits which was created by updating
+        source-git from dist-git (or vice versa) denoted by git trailers.
+
+        Determining the latest pair by commit time is not sufficient since
+        the datetime may be the same in all commits. We need to check for
+        actual parent-children relationships to determine the ordering.
+
+        Returns:
+            A tuple containing hash of the source-git commit and hash of the
+                corresponding dist-git commit. This represents the latest
+                update pair in both source-git and dist-git.
+
+        Raises:
+            PackitException, if no commits with git trailers that are used
+                for checking the sync could be found.
+        """
+        # Use source-git commit first in both tuples for consistency
+        sg_update_commits = [
+            (
+                c.hexsha,
+                re.search(rf"{re.escape(FROM_DIST_GIT_TOKEN)}: (.+)", c.message).group(
+                    1
+                ),
+            )
+            for c in self.up.local_project.get_commits(
+                max_count=1, grep=rf"^{re.escape(FROM_DIST_GIT_TOKEN)}: .\+$"
+            )
+        ]
+        dg_update_commits = [
+            (
+                re.search(
+                    rf"{re.escape(FROM_SOURCE_GIT_TOKEN)}: (.+)", c.message
+                ).group(1),
+                c.hexsha,
+            )
+            for c in self.up.local_project.get_commits(
```

Comment:
I would have expected to see `self.dg.local_project` here... so know I'm confused why the `up` is searched for both tokens... :thinking:

Reply:
Okaaaay... so now I'm wondering why I didn't find this issue while trying out the command :smile: